synapse/tests/test_lib_lmdbslab.py

```diff
@@ -397,3 +399,23 @@ def test_slab_math(self):
                 slab.dropdb('foo')
                 self.false(slab.dbexists('foo'))
                 self.gt(slab.mapsize, before_mapsize)
+
+class LmdbSlabMemLockTest(s_t_utils.SynTest):
+
+    async def test_lmdbslabmemlock(self):
+        self.thisHostMust(hasmemlocking=True)
+
+        beforelockmem = s_thisplat.getCurrentLockedMemory()
+
+        with self.getTestDir() as dirn:
+
+            path = os.path.join(dirn, 'test.lmdb')
+            async with await s_lmdbslab.Slab.anit(path, map_size=1000000, lockmemory=True):
+
+                if s_thishost.get('hasmemlocking'):
+                    for _ in range(3):
+                        lockmem = s_thisplat.getCurrentLockedMemory()
+                        if lockmem != beforelockmem:
+                            break
+                        await asyncio.sleep(1)
```

Comment:
Idea: If the lmdbslab had a second threading.Event we could clear it entering the lock loop, and set it upon completion. We could use it to signal the to the ioloop thread that we are done locking; and avoid this range(3) / sleep loop.

Reply:
Yes, and then we're injecting test code into production code.

Reply:
Done.